unstructured/ingest/connector/one_drive.py

```diff
@@ -0,0 +1,211 @@
+import json
+import os
+from dataclasses import dataclass, field
+from pathlib import Path
+from typing import TYPE_CHECKING
+from unstructured.file_utils.filetype import EXT_TO_FILETYPE
+from unstructured.ingest.interfaces import (
+    BaseConnector,
+    BaseConnectorConfig,
+    BaseIngestDoc,
+    StandardConnectorConfig,
+)
+
+from unstructured.ingest.logger import logger
+from unstructured.utils import requires_dependencies
+
+if TYPE_CHECKING:
+    from office365.onedrive.driveitems import driveItem
+
+
+MAX_MB_SIZE = 512_000_000
+
+@dataclass
+class SimpleOneDriveConfig(BaseConnectorConfig):
+
+    client_id: str
+    client_credential: str = field(repr=False)
+    user_pname: str 
+    tenant: str= field(repr=False)
+    authority_url: str = field(repr=False)
+    recursive: bool = False
+
+    def __post_init__(self):
+        if not (self.client_id and self.client_credential and self.user_pname):
+            raise ValueError(
+                'Please provide one of the following mandatory values:'
+                '\n-client_id\n-client_credential\n-user_principal_name'
+            )
+        self.token_factory = self._acquire_token
+
+    @requires_dependencies(["msal"])
+    def _acquire_token(self):
+        from msal import ConfidentialClientApplication
+        try:
+            app = ConfidentialClientApplication(
+                authority=f'{self.authority_url}/{self.tenant}',
+                client_id=self.client_id,
+                client_credential=self.client_credential)
+            token = app.acquire_token_for_client(scopes=["https://graph.microsoft.com/.default"])
+        except ValueError as exc:
+            logger.error("Couldn't set up credentials for OneDrive")
+            raise exc
+        return token
+
+@dataclass
+class OneDriveIngestDoc(BaseIngestDoc):
+
+    config: SimpleOneDriveConfig
+    file: "driveItem"
+
+    def __post_init__(self):
+        self.ext = ''.join(Path(self.file.name).suffixes)
+        if not self.ext:
+            raise ValueError(f"Unsupported file without extension. ")
+        
+        if self.ext not in EXT_TO_FILETYPE.keys():
+            raise ValueError(
+                f"Extension not supported. "
+                f"Value MUST be one of {', '.join([k for k in EXT_TO_FILETYPE if k is not None])}.",
+            )
+        self._set_download_paths()
+
+    def _set_download_paths(self) -> str:
+        """Sets download and output directories"""
+        dpath = Path(f'{self.standard_config.download_dir}')
+        opath = Path(f'{self.standard_config.output_dir}')
+
+        if pref := self.file.get_property('parentReference','').path.split(':')[-1]:
```

Comment:
nit: just call this parent_ref or parent_reference (pref could easily be read as preference)